airflow-core/src/airflow/cli/cli_config.py

```
@@ -789,6 +789,11 @@ def string_lower_type(val):
     ("option",),
     help="The option name",
 )
+ARG_CONFIG_HIDE_SENSITIVE = Arg(
+    ("--hide-sensitive",),
+    action="store_true",
+    help="Hide sensitive configuration values (passwords, keys, etc.) in the output.",
+)
```

Should we hide by default and have `--no-mask-sensitive` instead?

---

@uranusjr I think that will be a better option. Thus, the situation will be 
- CLI`* list` will not show any values
- flag like `--no-mask-sensitive` will not mask the sensitive values and show all the values.
Should we have another flag to show only the non-sensitive values?

---

We agreed on the devlist (and there was a lazy consensus on that) - see https://lists.apache.org/thread/3dhzqvpw4z2x0wfokpmdncsjwws86zcs point 5:


> 5) local CLI * list (connections, variables, config) only by default
returns "keys" - and it will only return values when `--show-values`
is passed as command line option (with clear comment in help that this
option **might** show sensitive data, also when we do `* list` command
without `--show-values` we emit stderr output explaining that
potentially sensitive data is hidden and you need to specify
`--show-values` to see them

---

Let's follow by the letter what we agreed. Thsi is not an opinion any more, this is something what we agreed to on devlist.

---

> Should we have another flag to show only the non-sensitive values?


I think we can add that one flag.